importer/harvester.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/python
+# -*- coding: utf-8  -*-
+"""
+Download and process KMB data.
+
+Download xml metadata about KMB images, and preprocess
+it as json files.
+Requires a settings.json file containing an API key
+and a list of keywords. Generates one json file per
+keywords.
+"""
+import requests
+from xml.dom.minidom import parseString
+import time
+
+import batchupload.common as common
+from importer.kmb_massload import parser
+
+SETTINGS = "settings.json"
+THROTTLE = 0.5
+LOGFILE = 'kmb_massloading.log'
+
+
+def load_settings(filename=SETTINGS):
+    """Load settings from file."""
+    return common.open_and_read_file(filename, as_json=True)
+
+
+def save_data(data, filename='kmb_data.json'):
+    """Dump data as json blob."""
+    common.open_and_write_file(filename, data, as_json=True)
+    print("Saved file: {}.".format(filename))
+
+
+def create_url(keyword, hits_limit, start_record, api_key):
+    """
+    Create url from which to download image metadata.
+
+    :param keyword: keyword to search for
+    :param hits_limit: how many hits per page
+    :param start_record: from which item to start
+    :param api_key: key to access API
+    """
+    keyword = requests.utils.quote(keyword)
+    url_base = ("http://kulturarvsdata.se/ksamsok/api?x-api={api_key}"
+                "&method=search&hitsPerPage={hits_limit}"
+                "&startRecord={start_record}"
+                "&query=serviceOrganization=RA%C3%84%20"
+                "and%20serviceName=KMB%20"
+                "and%20itemType=foto%20and%20mediaLicense=*%20"
+                "and%20text={keyword}")
+    return url_base.format(api_key=api_key,
+                           hits_limit=hits_limit,
+                           start_record=start_record,
+                           keyword=keyword)
+
+
+def split_records(dom):
+    """Split xml data into separate entry objects."""
+    return dom.getElementsByTagName("record")
+
+
+def parse_record(dom, record_dict, log):
+    """Parse and process the xml metadata into a dict."""
+    return parser(dom, record_dict, log)
+
+
+def get_records_from_url(url):
+    """Download xml metadata from url."""
+    with requests.get(url) as response:
+        source = response.text
+    return parseString(source)
+
+
+def get_records_from_file(filename):
+    """Get xml metadata from file, used for testing."""
+    with open(filename) as f:
+        data = "".join(line.rstrip() for line in f)
+    return parseString(data)
+
+
+def get_total_hits(records_blob):
+    """Extract total number of hits from xml metadata."""
+    hits_tag = records_blob.getElementsByTagName('totalHits')[0]
+    return int(hits_tag.firstChild.nodeValue)
+
+
+def extract_id_number(records_blob):
```

Comment:
Rename the parameter `record_blob` to clarify that it is for a single record (as opposed to `get_total_hits`) and add the parameter to the docstrings of both of these functions.